### ISSUE TITLE
tf.layers.batch_normalization missing moving variance and mean, low convolutional model accuracy

### ISSUE THREAD

**Issue by mateuszjurewicz** [tensorflow/tensorflow#27685]
**System information**
- I have written custom code
- Linux Ubuntu 16.04
- TensorFlow installed via pip from binary
- TensorFlow version: 1.7.0
- Python version: 3.6.3
- Bazel: 0.14.1
- GCC: 5.4.0
- CUDA 9.1.0, cuDNN: 7.1.2
- GPU model and memory: NVIDIA Quadro P6000/PCIe/SSE2, 24GB memory

**Describe the current behavior**
A custom, deep convolutional model, with batch normalization, does not converge. It is a recreated version of a Keras model that reaches accuracy of > 0.8 on the same data, whereas the tensorflow model built from tf.layers gets stuck under accuracy of < 0.3.

Presumed issue is lack of the moving mean and variance operation in the update ops, but all recommended methods of including them (https://towardsdatascience.com/pitfalls-of-batch-norm-in-tensorflow-and-sanity-checks-for-training-networks-e86c207548c8) have failed.

**Describe the expected behavior**

The model should converge to a much higher accuracy and the moving mean and variance operations should be visible in the update ops.

**Code to reproduce the issue**
I've created a github gist file that can be run on randomly generated data and prints reports about the ops:

https://gist.github.com/mateuszjurewicz/881d13e066f00b349b3eaafe901207f4

Usage:

python train_tensorflow.py --random_data=True

Alternatively the preprocessed data can be downloaded from:
https://archive.org/details/voice_data

You can also easily use it in combination with tensorboard, just ask the command argument parser for some --help.

**Other info / logs**
Larger context about this and the pure keras script that converges can be found here:
https://github.com/mateuszjurewicz/tensorflow_speech_recognition/blob/master/train_keras.py

**Comment by ymodak** [comment]
Apologies for the delay in response. Can you please try to replicate this issue with smaller example, possibly related to tf.layers.batch_normalization? It's difficult to debug larger models. Thanks!

**Comment by ymodak** [comment]
Automatically closing due to lack of recent activity. Please update the issue when new information becomes available, and we will reopen the issue. Thanks!

**Comment by mateuszjurewicz** [comment]
Hey, thank you for the response!
I don't think I can replicate this with a smaller model as the example is the only stage where the behavior occurs - a smaller model wouldn't converge with the Keras implementation, thus making the discrepancy between the Keras and pure Tensorflow disappear. I realize it can seem complex, I tried to make it as easy to run as possible. I understand if you can't diagnose it, but I'll appreciate any advice.
